Remotion/Web/UnitTests/Core/Resources/ResourcePathBuilderBasedStaticResourceCacheKeyProviderTest.cs

```diff
@@ -118,21 +118,48 @@ public void AppendResourcesFolderDetails ()
-      folder.CreateFile(@"Remotion.ObjectBinding.Web\Html\a\b\c\d\file3.txt", "määääääähhh", DateTime.FromFileTime(130000077777777777), DateTime.FromFileTime(130000088888888888));
+      folder.CreateFile(Path.Combine("Remotion.Web", "b.txt"));
+      folder.CreateFile(
+          Path.Combine("Remotion.Web", "Html", "file1.txt"),
+          "content",
+          DateTime.FromFileTime(130000011111111111),
+          DateTime.FromFileTime(130000022222222222));
+      folder.CreateFile(
+          Path.Combine("Remotion.Web", "Html", "sub", "file2.txt"),
+          "abc",
+          DateTime.FromFileTime(130000033333333333),
+          DateTime.FromFileTime(130000044444444444));
+      folder.CreateFile(Path.Combine("Remotion.Web", "UI", "c.txt"));
+      folder.CreateFile(Path.Combine("Remotion.Web", "UI", "sub", "d.txt"));
+      folder.CreateFile(
+          Path.Combine("Remotion.ObjectBinding.Web", "Html", "file3.txt"),
+          "defg",
+          DateTime.FromFileTime(130000055555555555),
+          DateTime.FromFileTime(130000066666666666));
+      folder.CreateFile(
+          Path.Combine("Remotion.ObjectBinding.Web", "Html", "a", "b", "c", "d", "file3.txt"),
+          "määääääähhh",
+          DateTime.FromFileTime(130000077777777777),
+          DateTime.FromFileTime(130000088888888888));
 
       using var staticResourceCacheKeyProvider = new TestStaticResourceCacheKeyProvider(folder.FullName, new[] { ResourceType.Html });
 
-      var expectedCacheKey = CreateExpectedCacheKeyForEntries(
-          (@"Remotion.ObjectBinding.Web\Html\file3.txt", 4, 130000066666666666),
-          (@"Remotion.ObjectBinding.Web\Html\a\b\c\d\file3.txt", 18, 130000088888888888),
-          (@"Remotion.Web\Html\file1.txt", 7, 130000022222222222),
-          (@"Remotion.Web\Html\sub\file2.txt", 3, 130000044444444444));
+      var expectedCacheKeyEntries = new[]
+                                    {
+                                        (Path.Combine("Remotion.ObjectBinding.Web", "Html", "file3.txt"), 4L, 130000066666666666),
+                                        (Path.Combine("Remotion.ObjectBinding.Web", "Html", "a", "b", "c", "d", "file3.txt"), 18L, 130000088888888888),
+                                        (Path.Combine("Remotion.Web", "Html", "file1.txt"), 7L, 130000022222222222),
+                                        (Path.Combine("Remotion.Web", "Html", "sub", "file2.txt"), 3L, 130000044444444444)
+                                    };
+
+      if (!OperatingSystem.IsWindows())
+      {
+        // On Linux the order the files are listed is different so we need to reorder the entries
+        (expectedCacheKeyEntries[0], expectedCacheKeyEntries[2]) = (expectedCacheKeyEntries[2], expectedCacheKeyEntries[0]);
+        (expectedCacheKeyEntries[1], expectedCacheKeyEntries[3]) = (expectedCacheKeyEntries[3], expectedCacheKeyEntries[1]);
+      }
```

Comment:
Something for future refernece: should we have computed the sha256 hash for each file and then XORed them isntead of putting them all in a giant memory stream? That way, we could have gotten commutative results and could have saved some memory allocations.